javaslang/src/main/java/javaslang/Lazy.java

```diff
@@ -135,14 +133,11 @@ private Lazy(Supplier<? extends T> supplier) {
      */
     @Override
     public T get() {
-        // using a local var speeds up the double-check idiom by 25%, see Effective Java, Item 71
-        Supplier<? extends T> tmp = supplier;
-        if (tmp != null) {
+        if ((value == null) && (supplier != null)) {
```

Comment:
@paplorinc we can do better ;) instead of

```java
CompletableFuture.allOf(
            values.flatMap(t -> range(0, 10)
                    .map(j -> runAsync(() -> assertThat(t._1.get()).isEqualTo(t._2)))
            ).toJavaArray(CompletableFuture.class)
    ).join();
```

we could use

```java
Future.sequence(
    values.flatMap(t -> range(0, 10)
                    .map(j -> runAsync(() -> assertThat(t._1.get()).isEqualTo(t._2)))
)/*.await()*/;
```